syntax = "proto3";

package bayes.bob;

message ContactLeadTemplate {
  // Short name of the type of contact lead. May contain template vars (see
  // populate_template on server code).
  string name = 1;

  // A list of filter IDs: do not present this lead if one these filters
  // returns false for the project. See the list of available filters in
  // scoring.py.
  repeated string filters = 2;

  // An example of email one should send to a contact to get their help. May
  // contain template vars (see populate_template on server code).
  string email_template = 3;

  // An extra tip on how to approach the lead, or send the email.
  string contact_tip = 4;

  // Text to display in the card before it's expanded if this lead is promoted
  // as the first thing we want to show.
  string card_content = 5;
}

message ContactLeads {
  // Sorted list of lead for user to contact and use or expand their network.
  repeated ContactLead leads = 1;
}

message ContactLead {
  // Short name of the type of contact lead.
  string name = 1;

  // An example of email one should send to a contact to get their help.
  string email_example = 2;

  // An extra tip on how to approach the lead, or send the email.
  string contact_tip = 3;
}
